pkg/crypto.go

```diff
@@ -440,29 +440,36 @@ func (client *Crypto) KeyExistsFor(legalEntity types.LegalEntity) bool {
 
 // VerifyWith verfifies a signature of some data with a given PublicKeyInPEM. It uses the SHA256 hashing function.
 func (client *Crypto) VerifyWith(data []byte, sig []byte, key jwk.Key) (bool, error) {
-	hashedData := sha256.Sum256(data)
-
 	mKey, err := key.Materialize()
 	if err != nil {
 		return false, err
 	}
-
-	if k, ok := mKey.(*rsa.PublicKey); ok {
-		if err := rsa.VerifyPKCS1v15(k, crypto.SHA256, hashedData[:], sig); err != nil {
+	var publicKey *rsa.PublicKey
+	if pk, ok := mKey.(*rsa.PublicKey); ok {
+		publicKey = pk
+	} else {
+		return false, ErrInvalidAlgorithm
+	}
+	// Verify as JWS if we can parse it as such
+	msg, _ := jws.Parse(bytes.NewReader(sig))
```

Comment:
any specific errors that we need to know about?

Reply:
for example it's a jws but wrongly formatted, then it would be validated as plain RSA, but error would set you on the wrong foot.

Reply:
We could check whether the input looks like a JWS by converting it to a string. If all characters are are alphanumeric (plus some extra chars like _- etc) and it contains dots (.) than it's probably a JWS. However, it could also be an RSA PKCS#1v15 signature which accidentally looks like an JWS leading to false negatives. I'd not take the chance.